packages/webpack/template-webpack-plugin/src/WebEncodePlugin.ts

```diff
@@ -87,7 +87,13 @@ export class WebEncodePlugin {
               styleInfo: encodeOptions['styleInfo'],
               manifest: encodeOptions.manifest,
               cardType: encodeOptions['cardType'],
-              pageConfig: encodeOptions.compilerOptions,
+              pageConfig: {
+                ...encodeOptions.compilerOptions,
+                ...((encodeOptions['sourceContent'] || {}) as Record<
```

Comment:
Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator.eslint[@typescript-eslint/prefer-nullish-coalescing](https://typescript-eslint.io/rules/prefer-nullish-coalescing)